framework/src/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java

```diff
@@ -361,7 +359,12 @@ private void clampToLowerBound(InferenceResult fromArgSupertypes, List<Annotated
         InferenceResult inferredFromArgEqualities = equalitiesSolver.solveEqualities(targets, argConstraints, typeFactory);
 
         Set<TypeVariable> remainingTargets =  inferredFromArgEqualities.getRemainingTargets(targets, true);
-        InferenceResult inferredFromLubs = supertypesSolver.solveFromArguments(remainingTargets, argConstraints, typeFactory);
+        InferenceResult inferredFromLubs = supertypesSolver.solveFromSupertypes(remainingTargets, argConstraints, typeFactory);
+
+        //TODO: solveFromSubtypes and then fromSubtypes (in infer) are misnamed if we use this method in this context
+        //TODO: find a clearer name
+        InferenceResult inferredFromGlbs = subtypesSolver.solveFromSubtypes(remainingTargets, argConstraints, typeFactory);
+        inferredFromLubs.mergeSubordinate(inferredFromGlbs);
 
```

Comment:
Looks good. Please do a search for similar code patterns to make sure they also handle both cases.